arista/cvp/plugins/modules/cv_configlet.py

```diff
@@ -226,14 +229,18 @@ def configlet_action(module):
                     else:
                         changed = True
                         deleted.append({configlet['name']: "success"})
+                        if 'taskIds' in delete_resp:
+                            for taskId in delete_resp['taskIds']:
+                                tasks.append(task_info(module=module, taskId=taskId))
```

Comment:
not sure it this warrants optimisation or not, but we could just append taskIds here and do a get_tasks at the end and filter out the ones with matching Ids. This would need 1/Nth number of API calls.

Reply:
It will be better to use make a single for loop to get all tasks information.

In term of number of API calls, it does not change a lot as every single task requires 1 API calls and API does not support `GET` for a list of tasks

As it will be easier to maintain, I am going to make this call with all tasks instead of per type of task. So if we have to update, it will be only in one place and not in 3 different blocs.

Reply:
i think there's a get_tasks call that can give you a full list of all pending tasks
https://github.com/aristanetworks/cvprac/blob/develop/cvprac/cvp_api.py#L138
so my idea was to collect all task Ids (whatever we recieve from CVP) and do the following at the very end of the module run
```
all_tasks = api.get_tasks()
taskList = [t for t in all_tasks if t["id"] in taskIds]
```

Reply:
but feel free to disregard this. i'm not sure it warrants the complexity introduced. your code looks very readable.

Reply:
Ok, So I would push the following code :

```python
# Collect task information for all generated tasks.
# Could be enhance with a call to get all pending tasks and then filtered by TaskId.
if len(taskList) > 0:
    for taskId in taskList:
        tasks.append(task_info(module=module, taskId=taskId))`
```

`taskList` would be populated with the following snippet:

```python
if 'taskIds' in new_resp:
    taskList = taskList + new_resp['taskIds']
```